### PR TITLE
Add various spelling corrections for encode and it's variants.

### PR DESCRIPTION
**Pull request by cfi-gb**
See e.g.

https://grep.app/search?q=enconded
https://grep.app/search?q=encocde
https://grep.app/search?q=enocding

and so on.

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -10264,10 +10264,16 @@ enclosue->enclosure
 enclosung->enclosing
 enclude->include
 encluding->including
+encocde->encode
 encocded->encoded
+enconde->encode
+enconded->encoded
 enconding->encoding
 encondings->encodings
+encorde->encode
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:10273]: Also suggest encore

Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: 446263d8d0351e5b4afa42831eab365e2a13b374

**Comment by cfi-gb** [comment]
> Can you also add the encodes version of the various typos

Good point, thanks again. Added these in 446263d8d0351e5b4afa42831eab365e2a13b374

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -10264,10 +10264,21 @@ enclosue->enclosure
...
 enconding->encoding
 encondings->encodings
+encorde->encode, encore,
 encorded->encoded
+encordes->encodes
+encording->encoding
+encordings->encodings
 encorporating->incorporating
 encose->enclose
 encosed->enclosed
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:10284]: I know it's not your change, but these, and the ones below, should probably also suggest encode*, given the s/d positioning on QWERTY keyboards.

Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: Good catch again, done in 6036275bfdff22c96381a7232759e4473fc9b134. Not sure if we should also add the following:

encosings->enclosings, encodings,

**Comment by cfi-gb** [comment]
> Sorry, I also missed it before, can you add the encoder and encoders variants of all these typos please.

Sure, done in e1046cde735ce6458c3d1ec6db2ffd1c459af4ab

> Also can you check the decode and equivalent variants are there too.

Done in 6036275bfdff22c96381a7232759e4473fc9b134

Hope i was able to catch / implement all suggestions correctly.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt]: Yeah nice spot!

Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: Added in fa0481780e35a11a3d01bbf8501b8bffb3cc68ff

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -7760,15 +7767,32 @@ decompressd->decompressed
...
+decorder->decoder
+decorders->decoders
+decordes->decodes
+decording->decoding
+decordings->decodings
 decorrellation->decorrelation
+decose->decode
+decosed->decoded
+decoses->decodes
+decosing->decoding
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:7795]: Add `decosings->decodings`

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -7760,15 +7767,32 @@ decompressd->decompressed
...
+decorde->decode
+decorded->decoded
+decorder->decoder
+decorders->decoders
+decordes->decodes
+decording->decoding
+decordings->decodings
 decorrellation->decorrelation
+decose->decode
+decosed->decoded
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:7793]: Add:
decoser->decoder
decosers->decoders

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -10264,15 +10294,33 @@ enclosue->enclosure
...
+enconder->encoder
+enconders->encoders
+encondes->encodes
 enconding->encoding
 encondings->encodings
+encorde->encode, encore,
 encorded->encoded
+encorder->encoder
+encorders->encoders
+encordes->encodes
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:10315]: Suggest encores too.

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -10264,15 +10294,33 @@ enclosue->enclosure
...
+encording->encoding
+encordings->encodings
 encorporating->incorporating
-encose->enclose
-encosed->enclosed
-encoses->encloses
-encosing->enclosing
+encose->enclose, encode,
+encosed->enclosed, encoded,
+encoses->encloses, encodes,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:10321]: Can we add:
encoser->encoder
encosers->encoders

You could also suggest encloser(s), but that seems to be fairly rare.

**Comment by cfi-gb** [comment]
Sure, will tackle this next week. Thanks again for the review.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:7797]: 8f0960943d8c462b58cbce05241c261c365de429

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:7793]: 8f0960943d8c462b58cbce05241c261c365de429

File: codespell_lib/data/dictionary.txt (hunk 5 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: 8f0960943d8c462b58cbce05241c261c365de429

File: codespell_lib/data/dictionary.txt (hunk 6 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:10326]: 8f0960943d8c462b58cbce05241c261c365de429

**Comment by cfi-gb** [comment]
Also found one single occurrence of the following correction which i have extended in 581accedc0fdda8039746d970fd4a2d5b1f82240

```
endcoding->encoding
```

**Comment by cfi-gb** [comment]
Added transcode* and it's variants in b6287eaa94eb6879d1fb727b39e144efc2396570, hope i got everything right because the PR is now quite huge. 

I have left out the following for now because i wasn't sure if there variants of transcode for this typo:

```
enocde->encode
```

**Comment by peternewman** [comment]
> I have left out the following for now because i wasn't sure if there variants of transcode for this typo:
> 
> ```
> enocde->encode
> ```

`s/en/trans/g` i.e. transocde, which grep.app has a few hits for:
https://grep.app/search?q=transocde&words=true

**Comment by cfi-gb** [comment]
Ah, indeed. Thanks again, have added these variants in 0adee970c40c399c843c0f3ee2d7fc1f46431279

**Pull request merged**